### PR TITLE
refactor: replace `Supabase` button with Native Button 

### LINKED ISSUE #854: feature: Add loading state to our buttons to give users feedback when an action is loading
@OgDev-01 Please assist in provide examples. Should this be applied to all buttons?

### LINKED ISSUE #886: feat: Make highlights editable
As a user I should be able to edit a highlight. 

Screen Shot 2023-02-20 at 9 35 07 AM

https://www.figma.com/file/OpVX6WT7dmWqnwRuEvADMF/OpenSauced-%E2%80%94-Insights-(insights.opensauced.pizza)?node-id=5876%3A7583&t=KG4cKsqY6B6hvToP-0

### LINKED ISSUE #901: refactor: replace current Supabase `Button` component with radix
This has been discussed [here](https://github.com/open-sauced/insights/issues/338)

Radix currently has no button, But we can leverage  https://ui.shadcn.com/docs/primitives/button. 

### Additional Context

The button should have a loading state also https://github.com/open-sauced/insights/issues/854. Like a spinner when an action is happening asynchronously.

### PR DESCRIPTION
**Pull request by OgDev-01**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description
This PR replaces the Supabase Ui button with the native button component and adds a loading state to the button component.

### Note:
This was built with accessibility support in mind and currently supports all accessibility attributes of the Native button element.

## Related Tickets & Documents
Fixes #901 
Closes #854 

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Review by 0-vortex — COMMENTED** [review]
Don't know what the PR is about but good button fix!!!

Know it's a draft but tooling wise this is what I can see:
1. some important classes in the `Button` usage are not needed anymore
2. some code refactors look like custom eslint/prettier configuration

**Comment by bdougie** [comment]
@OgDev-01 I appreciate you attacking the refactor, but this PR is confusing to review. The edit feature is sandwiched with the refactor. My suggestion would be making this the refactor PR and do the edit feature in a new PR.

**Comment by OgDev-01** [comment]
> @OgDev-01 I appreciate you attacking the refactor, but this PR is confusing to review. The edit feature is sandwiched with the refactor. My suggestion would be making this the refactor PR and do the edit feature in a new PR.

Ok

File: components/atoms/Button/button.tsx [hunk 1]
```diff
@@ -1,34 +1,89 @@
 import React from "react";
-import { Button as SupabaseButtonComponent } from "@supabase/ui";
+import clsx from "clsx";
+// import { Button as SupabaseButtonComponent } from "@supabase/ui";
```
Comment (bdougie) [inline components/atoms/Button/button.tsx:3]: Remove

File: components/atoms/Button/button.tsx [hunk 2]
```diff
@@ -1,34 +1,89 @@
...
 
-interface ButtonProps extends React.ComponentProps<typeof SupabaseButtonComponent> {
-  //Add additional prop definitions here
+// interface ButtonProps extends React.ComponentProps<typeof SupabaseButtonComponent> {
+//   //Add additional prop definitions here
+// }
```
Comment (bdougie) [inline components/atoms/Button/button.tsx:7]: Removed dead code

**Review by bdougie — APPROVED** [review]

**Review by brandonroberts — APPROVED** [review]
Code LGTM. Merge conflicts need resolving

**Pull request merged**